onnxruntime/core/providers/cuda/nn/conv.cc

```diff
@@ -166,40 +166,40 @@ Status Conv<T>::ComputeInternal(OpKernelContext* context) const {
       s_.algo = perf.algo;
       s_.workspace_bytes = perf.memory;
     }
-  }
```

Comment:
Move this brace down until after all CuDNN calls and effectively hold onto the mutex until it is safe to be released